tools/check-magic-strings.sh

```diff
@@ -20,13 +20,18 @@ MAGIC_STRINGS_TEMP=`mktemp lit-magic-strings.inc.h.XXXXXXXXXX`
 
 cp $MAGIC_STRINGS_INC_H $MAGIC_STRINGS_TEMP
 $MAGIC_STRINGS_GEN
-diff -q $MAGIC_STRINGS_INC_H $MAGIC_STRINGS_TEMP
-DIFF_RESULT=$?
-mv $MAGIC_STRINGS_TEMP $MAGIC_STRINGS_INC_H
 
-if [ $DIFF_RESULT -ne 0 ]
-then
-  echo -e "\e[1;33m$MAGIC_STRINGS_INC_H must be re-generated. Run $MAGIC_STRINGS_GEN\e[0m"
+if [ $? -eq 0 ]; then
+  diff -q $MAGIC_STRINGS_INC_H $MAGIC_STRINGS_TEMP
+  DIFF_RESULT=$?
+  mv $MAGIC_STRINGS_TEMP $MAGIC_STRINGS_INC_H
```

Comment:
This `mv` should happen no matter what (as it restores the potentially scratched or overwritten header file), so it should be moved after `fi`.